Add Pressure and Altitude discovery schemas for Matter Eve Weather device

lboue opened the pull request:
## Breaking change

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
- Add Pressure and Altitude attributes for EveWeather device


Altitude
` "1/319486977/319422483": 40.0,
`

Pressure
` "1/319486977/319422484": 1008.5,
`
## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->


- This PR fixes or closes issue: 
- This PR is related to issue: 
- This PR is related to PR: [Add Eve Weather custom attributes #842](https://github.com/home-assistant-libs/python-matter-server/pull/842) and [#855](https://github.com/home-assistant-libs/python-matter-server/pull/855).
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

![image](https://github.com/user-attachments/assets/d5103086-012e-42cb-a83b-0d32ddd72e99)

Local testing:

![image](https://github.com/user-attachments/assets/3606cd1c-b739-4ae1-9c8e-13ff62a741e9)

lboue committed:
Update number.py to add EveWeatherAltitude attribute

lboue committed:
Update sensor.py to add EveCluster Pressure Attribute

lboue committed:
Update strings.json

lboue committed:
Create eve-weather-sensor.json

lboue committed:
Update test_sensor.py

lboue committed:
Update eve-weather-sensor.json

lboue committed:
Update test_sensor.py

Pressure
AttributeId: 319422484 (0x00130a0014) - Value type: float32

lboue commented:
I don't understand why testing can't find the Pressure attribute.
```
Python 3.12.6
Error: homeassistant/components/matter/sensor.py:249: error: "type[Attributes]" has no attribute "Pressure"  [attr-defined]
Error: homeassistant/components/matter/number.py:154: error: "type[Attributes]" has no attribute "Altitude"  [attr-defined]
Found 2 errors in 2 files (checked 24 source files)
Error: Process completed with exit code 1.
```

It is defined here:

https://github.com/lboue/python-matter-server/blob/main/matter_server/common/custom_clusters.py#L269-L288

lboue committed:
Update test_sensor.py

lboue committed:
Update test_sensor.py

lboue committed:
Update test_sensor.py

TheJulianJES commented:
It's likely not included in the library version that HA uses at the moment: https://github.com/home-assistant/core/blob/dev/homeassistant/components/matter/manifest.json
It should probably be bumped to 6.5.0.

lboue committed:
Update manifest.json

Bump to python-matter-server==6.5.0

lboue commented:
> It's likely not included in the library version that HA uses at the moment: https://github.com/home-assistant/core/blob/dev/homeassistant/components/matter/manifest.json It should probably be bumped to 6.5.0.

Thanks. I changed it in the manifest.json file but it still fails.

marcelveldt commented:
> Thanks. I changed it in the manifest.json file but it still fails.

Did you run HA once so it installed the version you specified in the manifest ?

lboue committed:
Update requirements_all.txt

Bump requirements to python-matter-server 6.5.0

lboue committed:
Update requirements_test_all.txt

Bump requirements to python-matter-server 6.5.0

lboue commented:
Now I only have 1 failing checks
```
=========================== short test summary info ============================
ERROR tests/components/matter/test_sensor.py::test_eve_weather_sensor_custom_cluster[True] - AttributeError: 'NoneType' object has no attribute 'items'
Error: Process completed with exit code 1.
```

lboue commented:
> > Thanks. I changed it in the manifest.json file but it still fails.
> 
> Did you run HA once so it installed the version you specified in the manifest ?

Fixed

lboue committed:
Update test_sensor.py

lboue commented:
Someone could help me to fix this error?

`ERROR tests/components/matter/test_sensor.py::test_eve_weather_sensor_custom_cluster[True] - AttributeError: 'NoneType' object has no attribute 'items'`

lboue committed:
Update test_sensor.py

lboue committed:
Update sensor.py

lboue committed:
Update sensor.py

lboue committed:
Update test_sensor.py

lboue committed:
Update sensor.py

lboue committed:
Update test_sensor.py

lboue committed:
Update test_sensor.py

lboue committed:
Update test_sensor.py

lboue commented:
Local testing:

![image](https://github.com/user-attachments/assets/3606cd1c-b739-4ae1-9c8e-13ff62a741e9)

marcelveldt commented on homeassistant/components/matter/manifest.json:
```diff
@@ -6,6 +6,6 @@
   "documentation": "https://www.home-assistant.io/integrations/matter",
   "iot_class": "local_push",
-  "requirements": ["python-matter-server==6.3.0"],
+  "requirements": ["python-matter-server==6.5.0"],
```
```suggestion
  "requirements": ["python-matter-server==6.3.0"],
```

marcelveldt commented on homeassistant/components/matter/manifest.json:
Please leave the dependency bump out of this PR.
There is a PR that bumps the library: https://github.com/home-assistant/core/pull/126636

Once that is merged, I'll rebase this PR and fix that last error.

marcelveldt commented:
> Someone could help me to fix this error?
> 
> `ERROR tests/components/matter/test_sensor.py::test_eve_weather_sensor_custom_cluster[True] - AttributeError: 'NoneType' object has no attribute 'items'`

For next time, you need to look carefully at the format of the node fixtures. Its not exactly the same as the diagnostics dump. Basically its only the "node" section of the diagnostics dump.

marcelveldt committed:
fix test fixture

marcelveldt committed:
Update requirements_all.txt

marcelveldt committed:
Update requirements_test_all.txt

marcelveldt committed:
Update manifest.json

lboue commented:
> For next time, you need to look carefully at the format of the node fixtures. Its not exactly the same as the diagnostics dump. Basically its only the "node" section of the diagnostics dump.

Is there a script to extract it or do you have to do it by hand?

lboue committed:
Merge branch 'dev' into EveWeather_custom_attributes

marcelveldt committed:
fix tests

marcelveldt committed:
Merge branch 'EveWeather_custom_attributes' of https://github.com/lboue/ha_core into pr/125690

marcelveldt commented:
> Is there a script to extract it or do you have to do it by hand?

Just do it by hand. You simply move the "node" section one level up and remove the rest.

I have now adjusted this PR for you so for this PR its not needed

marcelveldt committed:
Update test_sensor.py

marcelveldt reviewed (approved):
Approved from my PoV but can use an extra pair of eyes as I finished this PR's tests

joostlek commented on homeassistant/components/matter/number.py:
```diff
@@ -137,4 +138,19 @@ def _update_from_device(self) -> None:
     ),
+    MatterDiscoverySchema(
+        platform=Platform.NUMBER,
+        entity_description=MatterNumberEntityDescription(
+            key="EveWeatherAltitude",
+            entity_category=EntityCategory.CONFIG,
+            translation_key="altitude",
+            native_max_value=9000,
+            native_min_value=0,
+            native_unit_of_measurement=UnitOfLength.METERS,
+            native_step=1,
+            mode=NumberMode.BOX,
+        ),
```
no device class?

marcelveldt replied:
added

marcelveldt committed:
add device class

MartinHjelmare commented on tests/components/matter/test_sensor.py:
```diff
@@ -92,6 +92,16 @@ async def eve_energy_plug_patched_node_fixture(
 
 
+@pytest.fixture(name="eve_weather_sensor_node")
+async def eve_weather_sensor_node_fixture(
```
Please don't duplicate the same fixture in different modules. Put it in the `conftest.py` module instead. Those fixtures can be accessed by the whole test package.
